### DISCUSSION TITLE
Specification for feed icon

### DISCUSSION THREAD

**Discussion by AlphaJack** [spacecowboy/Feeder#270] [Q&A] [answered]
Hi, what are the requirements (tag location, image format, image limits) to have an icon next to the feed name?

I tried to set both `<icon>` and `<logo>` on:

- both at the beginning of the file and on a item basis,
- with the absolute URL of "favicon.svg"
- with the absolute URL of "favicon.ico" and "favicon.png", both sized 128x128px

but I could not suceed, as my website always display a black square.

<https://news.nononsenseapps.com/index.atom> uses

```
<icon>https://news.nononsenseapps.com/favicon.ico</icon>
<logo>https://news.nononsenseapps.com/apple-touch-icon.png</logo>
```

before the first entry

**Comment by spacecowboy** [comment]
I can answer a bunch of that but I think it is much easier if you just show me your feed and I'll tell you what you did wrong.

**Comment by AlphaJack** [comment, marked as answer]
I tried implemeting the Atom feed from scratch and it worked, and I realized I was mixing the syntax with the RSS 2.0 feed. 

Below the working code for both feeds, featuring website images (128x128 .ico) and post thumbnails (1200x630 .jpeg):

Atom 1.0
```xml
<?xml version="1.0" encoding="utf-8"?>
<feed xmlns="http://www.w3.org/2005/Atom" xml:lang="en">
  <id>https://example.org/</id>
  <title>Example title</title>
  <subtitle>Example description</subtitle>
  <link href="https://example.org/" rel="alternate" type="text/html" hreflang="en"/>
  <link href="https://example.org/blog/feed.atom" rel="self" type="application/atom+xml"/>
  <icon>https://example.org/favicon.png</icon>
  <logo>https://example.org/landscape-2-1-ratio.png</logo>
  <updated>2024-04-21T13:26:10.576Z</updated>
  <author>
    <name>First Last</name>
    <email>first.last@example.org</email>
  </author>
  
  <entry>
    <id>https://example.org/blog/post-1/</id>
    <title>Post title</title>
    <link href="https://example.org/blog/post-1/"/>
    <published>2023-03-04T00:00:00.000Z</published>
    <updated>2024-01-01T00:00:00.000Z</updated>
    <media:thumbnail xmlns:media="http://search.yahoo.com/mrss/" url="https://example.org/images/post-1.jpeg" />"
    <summary>Post excerpt</summary>
    <author>
      <name>First Last</name>
      <email>first.last@example.org</email>
    </author>
  </entry>

</feed>

```

RSS 2.0
```xml
<?xml version="1.0" encoding="UTF-8"?>
<rss version="2.0" xmlns:media="http://search.yahoo.com/mrss/" xmlns:atom="http://www.w3.org/2005/Atom">
  <channel>
    <title>Example title</title>
    <description>Example desc</description>
    <link>https://example.org/</link>
    <atom:link href="https://example.org/blog/feed.rss" rel="self" type="application/rss+xml"/>
    <image>
      <link>https://example.org/</link>
      <title>Example title</title>
      <description>Example desc</description>
      <url>https://example.org/favicon.png</url>
      <width>128</width>
      <height>128</height>
    </image>

    <item>
      <title>Post title</title>
      <link>https://example.org/blog/post-1/</link>
      <guid isPermaLink="true">https://example.org/blog/post-1/</guid>
      <description>Post Excerpt</description>
      <pubDate>Mon, 01 Jan 2024 00:00:00 GMT</pubDate>
      <media:content type="image/jpeg" width="1200" height="630" medium="image" url="https://example.org/images/post-1.jpeg"/>
      <author>first.last@example.org (First Last)</author>
    </item>

  </channel>
</rss>
```

**Reply by spacecowboy** [reply]
When you mentioned that `<icon>` wasn't working I suspected you were mixing up RSS and Atom.

I can mention that when a feed doesn't have a an image, then Feeder will download the main website and see if it defines a favicon or apple touch icon.

**Comment by AlphaJack** [comment]
@spacecowboy  I noticed that Feeder is cropping `<logo>` as the squared feed icon. Is it desired? By the specs, `<icon>` should be squared and `<logo>` should be 2:1

https://validator.w3.org/feed/docs/atom.html

**Reply by spacecowboy** [reply]
All feed icons will be shown in a square

**Reply by AlphaJack** [reply]
@spacecowboy  Ok, but why is Feeder using `<logo>` (landscape) instead  of `<icon>` (squared)?

**Reply by spacecowboy** [reply]
That choice comes from the gofeed library.
